earth_enterprise/rpms/opengee-fusion/pre-install.sh

```diff
@@ -68,18 +71,16 @@ END
     fi
 }
 
-create_users_and_groups()
-{
-    # Add user if it does not exist:
-    if [ -z "$(getent passwd "$GEFUSIONUSER")" ]; then
-		mkdir -p "$BASEINSTALLDIR_OPT/.users/$GEFUSIONUSER"
-		useradd --home "$BASEINSTALLDIR_OPT/.users/$GEFUSIONUSER" \
-            --system --gid "$GEGROUP" "$GEFUSIONUSER"
-        keyvalue_file_set "$GEE_INSTALL_KV_PATH" gefusionuser_existed "false"
```

Comment:
`gefusionuser_existed` is still used in `install_or_upgrade_asset_root()`. I'm not sure replacing this with `check_username` is going to work there.

Reply:
well, that's why it's a wip...I must have missed that.

Reply:
I believe that is now tested for differently, removing that requirement from the script.

Reply:
You replaced in one place, but not in another, so it's still not removed as a requirement.

Also, can you describe what the tests looking for whether `gegroup` and `gefusionuser` are trying to determine, and how you're trying to determine that without the need for checking for `gegroup` and `gefusionuser`?  I don't quite understand what we're looking for, and what the difference between an existing and a new installation of Fusion and Server are.  There's possibly other information we can use, like checking whether there exist files to upgrade, or whether we're installing a new RPM package, or upgrading an existing one.

Reply:
If we really need a strong determinant on whether we are doing a new vs upgrade install, there also is a mechanism for doing this by examining the $1 argument of the pre/post scripts, per:
  
https://fedoraproject.org/wiki/Packaging:Scriptlets

Debian pre/post install scripts support something somewhat similar, but I think passes the version of the old package if there is one, and the version of the current package, which can then be compared.

Reply:
I think that may not be the appropriate determinant we want.  I think a lot of this code was added to make the Open GEE 5.2 installers work with GEE installations that existed before open sourcing and used different installers.